net-misc/kea/kea-2.6.3.ebuild

```diff
@@ -0,0 +1,242 @@
+
+		sed -e '/^TEST_F(.*Sql.*) {$/,/^}$/d' \
+			-i src/lib/dhcpsrv/tests/alloc_engine4_unittest.cc || die
+
+		sed -e '/^TEST_F(Dhcpv.*SrvTest, checkConfigFiles) {$/,/^}$/d' \
+			-i src/bin/dhcp*/tests/dhcp*_srv_unittest.cc || die
+
+		sed -e '/^TEST_F(.*SQLTest,.*) {$/,/^}$/d' \
+			-i src/bin/dhcp4/tests/dora_unittest.cc || die
+
+		sed -e '/^TEST_F(DeclineTest,.*SQL) {$/,/^}$/d' \
+			-i src/bin/dhcp*/tests/decline_unittest.cc || die
+	fi
+
+	# do not create /run
+	sed -e '/$(DESTDIR)${runstatedir}/d' \
+		-i Makefile.am || die
+
+	eautoreconf
+}
+
+src_configure() {
+	# -Werror=odr
+	# https://bugs.gentoo.org/861617
+	#
+	# I would truly love to submit an upstream bug but their self-hosted gitlab
+	# won't let me sign up. -- Eli
+	filter-lto
+
+	local myeconfargs=(
+		--disable-install-configurations
+		--disable-rpath
+		# manually installed
+		--disable-shell
+		--disable-static
+		--enable-generate-messages
+		--enable-logger-checks
+		--enable-perfdhcp
+		--localstatedir="${EPREFIX}/var"
+		--runstatedir="${EPREFIX}/run"
+		--without-werror
+		--with-log4cplus
+		$(use_enable debug)
+		$(use_enable doc generate-docs)
+		$(use_with mysql)
+		$(use_with openssl)
+		$(use_with postgres pgsql)
+		$(usev test --with-gtest="${EPREFIX}/usr")
+	)
+	econf "${myeconfargs[@]}"
+}
+
+install_shell() {
+	python_domodule src/bin/shell/*.py
+	python_doscript src/bin/shell/kea-shell
+
+	# fix path to import kea modules
+	sed -e "/^sys.path.append/s|(.*)|('$(python_get_sitedir)/${PN}')|"	\
+		-i "${ED}"/usr/lib/python-exec/${EPYTHON}/kea-shell || die
+}
+
+src_install() {
+	emake -j1 install DESTDIR="${D}"
+
+	if use shell; then
+		python_moduleinto ${PN}
+		python_foreach_impl install_shell
+	fi
+
+	dodoc -r doc/examples
+
+	rm -f "${ED}"/usr/share/doc/${P}/COPYING
+
+	diropts -m 0750 -o root -g dhcp
+	dodir /etc/kea
+	insopts -m 0640 -o root -g dhcp
+	insinto /etc/kea
+	newins doc/examples/agent/comments.json kea-ctrl-agent.conf.sample
+	newins doc/examples/kea6/simple.json kea-dhcp6.conf.sample
+	newins doc/examples/kea4/single-subnet.json kea-dhcp4.conf.sample
+	newins doc/examples/ddns/comments.json kea-dhcp-ddns.conf.sample
+
+	# set log to syslog by default
+	sed -e 's/"output": "stdout"/"output": "syslog"/' \
+		-i "${ED}"/etc/kea/*.conf.sample || die
+
+	newconfd "${FILESDIR}"/${PN}-confd-r2 ${PN}
+	newinitd "${FILESDIR}"/${PN}-initd-r2 ${PN}
+
+	systemd_dounit "${FILESDIR}"/${PN}-ctrl-agent.service-r2
+	systemd_dounit "${FILESDIR}"/${PN}-dhcp-ddns.service-r2
+	systemd_dounit "${FILESDIR}"/${PN}-dhcp4.service-r2
+	systemd_dounit "${FILESDIR}"/${PN}-dhcp6.service-r2
+
+	newtmpfiles "${FILESDIR}"/${PN}.tmpfiles.conf ${PN}.conf
+
+	keepdir /var/lib/${PN} /var/log/${PN}
+	fowners -R dhcp:dhcp /var/lib/${PN} /var/log/${PN}
+	fperms 750 /var/lib/${PN} /var/log/${PN}
+
+	find "${ED}" -type f -name "*.la" -delete || die
+}
+
+pkg_postinst() {
+	tmpfiles_process ${PN}.conf
+
+	if ver_replacing -lt 2.6; then
```

Comment:
We may want a news item for this, but then again, it was buggered before. Open to debate on that.

Reply:
As you want, but I have any experience for news item (apart reading of course).

Reply:
It could be a good opportunity to get you such experience, but on the other hand, Kea has no stable keywords and it's completely broken in-tree (and has vulnerabilities, albeit not really affecting us so much). Another time.